Shared/Samples/Trace utility network/TraceUtilityNetworkView.Model.swift

```diff
@@ -0,0 +1,357 @@
+        func identifyFeatureAt(_ screenPoint: CGPoint, with proxy: MapViewProxy) async -> ArcGISFeature? {
+            try? await proxy.identifyLayers(
+                screenPoint: screenPoint,
+                tolerance: 10
+            ).first?.geoElements.first as? ArcGISFeature
+        }
+        
+        /// Sets the pending trace parameters with the provided trace type.
+        /// - Parameter type: The trace type.
+        func setTraceParameters(ofType type: UtilityTraceParameters.TraceType) {
+            pendingTraceParameters = UtilityTraceParameters(
+                traceType: type,
+                startingLocations: []
+            )
+            pendingTraceParameters?.traceConfiguration = mediumVoltageRadial?.defaultTraceConfiguration
+            tracingActivity = .settingPoints(pointType: .start)
+        }
+        
+        /// Resets all of the important stateful values for when a trace is cancelled or completed.
+        func reset() {
+            map.operationalLayers.forEach { ($0 as? FeatureLayer)?.clearSelection() }
+            points.removeAllGraphics()
+            pendingTraceParameters = nil
+            tracingActivity = .none
+        }
+        
+        /// Performs important tasks including adding credentials, loading and adding operational layers.
+        func setup() async {
+            do {
+                try await ArcGISEnvironment.authenticationManager.arcGISCredentialStore.add(.publicSample)
+                try await network?.load()
+            } catch {
+                await updateUserHint(withMessage: "An error occurred while loading the network.")
+                return
+            }
+            featureLayerURLs.forEach { url in
+                let table = ServiceFeatureTable(url: url)
+                let layer = FeatureLayer(featureTable: table)
+                if table.serviceLayerID == 3 {
+                    layer.renderer = UniqueValueRenderer(
+                        fieldNames: ["ASSETGROUP"],
+                        uniqueValues: [.lowVoltage, .mediumVoltage],
+                        defaultSymbol: SimpleLineSymbol()
+                    )
+                }
+                map.addOperationalLayer(layer)
+            }
+        }
+        
+        /// Runs a trace with the pending trace configuration and selects features in the map that
+        /// correspond to the element results.
+        ///
+        /// - Note: Elements are grouped by network source prior to selection so that all selections
+        /// per operational layer can be made at once.
+        func trace() async throws {
+            guard let pendingTraceParameters = pendingTraceParameters else { return }
+            let traceResults = try await network?.trace(using: pendingTraceParameters)
+                .filter { $0 is UtilityElementTraceResult }
+            for result in traceResults as? [UtilityElementTraceResult] ?? [] {
```

Comment:
I cannot think of a way to write it more concisely, so I blame Swift not being expressive enough! 😂

Reply:
Yes, this is cleaner, thank you!